qiskit/pulse/instruction_schedule_map.py

```diff
@@ -161,39 +169,89 @@ def get(self,
-            return schedule_generator(*params, **kwparams)
-        # otherwise this is just a Schedule
-        return schedule_generator
+        # callback function
+        if callable(schedule_param_tuple.schedule):
+            return schedule_param_tuple.schedule(**bind_parameters)
+
+        # schedule
+        sched_params = list(schedule_param_tuple.schedule.parameters)
+        if sched_params:
+            parameter_mapping = dict()
+            for sched_param in sched_params:
+                bind_value = bind_parameters[sched_param.name]
+                # if value is not set, keep the parameter unassigned
+                if bind_value is not None:
+                    parameter_mapping[sched_param] = bind_value
+            return schedule_param_tuple.schedule.assign_parameters(parameter_mapping)
+
+        return schedule_param_tuple.schedule
 
     def add(self,
             instruction: Union[str, Instruction],
             qubits: Union[int, Iterable[int]],
-            schedule: Union[Schedule, Callable[..., Schedule]]) -> None:
+            schedule: Union[Schedule, Callable[..., Schedule]],
+            params: Optional[List[str]] = None) -> None:
         """Add a new known instruction for the given qubits and its mapping to a pulse schedule.
 
         Args:
             instruction: The name of the instruction to add.
             qubits: The qubits which the instruction applies to.
             schedule: The Schedule that implements the given instruction.
+            params: List of parameter names to create a parameter-bound schedule from the
+                associated gate instruction. If :py:meth:`get` is called with arguments rather
+                than keyword arguments, this parameter list is used to map the input arguments to
+                parameter objects stored in the target schedule.
 
         Raises:
             PulseError: If the qubits are provided as an empty iterable.
         """
         instruction = _get_instruction_string(instruction)
 
+        # validation of input data
         qubits = _to_tuple(qubits)
         if qubits == ():
             raise PulseError("Cannot add definition {} with no target qubits.".format(instruction))
         if not (isinstance(schedule, Schedule) or callable(schedule)):
             raise PulseError('Supplied schedule must be either a Schedule, or a '
                              'callable that outputs a schedule.')
-        self._map[instruction][qubits] = schedule
+
+        # initialize parameter list
+        if isinstance(schedule, Callable):
+            func_parameters = list(inspect.signature(schedule).parameters.keys())
+        else:
+            func_parameters = set(param.name for param in schedule.parameters)
+
+        if params is None:
+            if len(func_parameters) > 0:
+                # for backward compatibility
+                params = sorted(func_parameters)
+            else:
+                params = []
```

Comment:
`sorted([])` works fine
```suggestion
            # for backward compatibility
            params = sorted(func_parameters)
```